### PR TITLE
feat(buildd): add support for ubuntu 23.10

### PR DESCRIPTION
**Pull request by syu-w**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

CRAFT-1759

### CODE REVIEW

File: craft_providers/multipass/multipass_provider.py [hunk 1]
```diff
@@ -100,8 +100,9 @@ def name(self) -> str:
     ubuntu.BuilddBaseAlias.KINETIC: RemoteImage(
         remote=Remote.RELEASE, image_name="kinetic"
     ),
-    ubuntu.BuilddBaseAlias.LUNAR: RemoteImage(
-        remote=Remote.RELEASE, image_name="lunar"
+    ubuntu.BuilddBaseAlias.LUNAR: RemoteImage(remote=Remote.DAILY, image_name="lunar"),
```
Comment (mr-cal) [inline craft_providers/multipass/multipass_provider.py:103]: Why did you change lunar from `release` to `daily`?

Comment (syu-w) [reply craft_providers/multipass/multipass_provider.py:103]: Does it have the same issue as buildd image or not?

Comment (mr-cal) [reply craft_providers/multipass/multipass_provider.py:103]: It shouldn't.  

The Multipass `daily` and `release` remotes are for "regular" cloud images and the `snapcraft` remote are for buildd images.

Comment (syu-w) [reply craft_providers/multipass/multipass_provider.py:103]: So we should use `release` for now and move to `snapcraft` later?

Comment (mr-cal) [reply craft_providers/multipass/multipass_provider.py]: Correct

Comment (syu-w) [reply craft_providers/multipass/multipass_provider.py]: It has been updated.

Comment (syu-w) [reply craft_providers/multipass/multipass_provider.py]: I don't think we are going to use mantic for a while.

File: tests/integration/multipass/test_multipass_provider.py [hunk 2]
```diff
@@ -51,6 +51,12 @@ def test_launched_environment(alias, installed_multipass, instance_name, tmp_pat
     if sys.platform == "darwin" and alias == BuilddBaseAlias.KINETIC:
         pytest.skip(reason="previous interim releases are not available on MacOS")
 
+    if sys.platform == "darwin" and alias == BuilddBaseAlias.LUNAR:
+        pytest.skip(reason="Lunar on MacOS is not available")
```
Comment (mr-cal) [inline tests/integration/multipass/test_multipass_provider.py:55]: The release image of Lunar should be testable on macos too.

**Comment by syu-w** [comment]
currently blocked by buildd.

**Comment by jugmac00** [comment]
Hi friends, we got an inquiry about adding support for mantic to LPCI ( https://bugs.launchpad.net/lpci/+bug/2038812 ).

When do you plan to add support for mantic in craft-providers?

Cheers,
Jürgen

**Comment by mr-cal** [comment]
Looks like mantic is now available as a daily buildd image, so we should be unblocked and able to merge this.

File: tests/integration/multipass/test_multipass_provider.py [hunk 3]
```diff
@@ -52,6 +52,12 @@ def test_create_environment(installed_multipass, instance_name):
 def test_launched_environment(alias, installed_multipass, instance_name, tmp_path):
     """Verify `launched_environment()` creates and starts an instance then stops
     the instance when the method loses context."""
+    if sys.platform == "darwin" and alias == BuilddBaseAlias.KINETIC:
+        pytest.skip(reason="previous interim releases are not available on MacOS")
+
+    if sys.platform == "darwin" and alias == BuilddBaseAlias.MANTIC:
+        pytest.skip(reason="Mantic on MacOS is not available")
```
[the remark this answers is not in the document]
Comment (syu-w) [reply tests/integration/multipass/test_multipass_provider.py:59]: It now supported?

**Comment by freyes** [comment]
The failure in the CI job doesn't seem to be related to this patch

```
  multipass/test_multipass_provider.py::test_launched_environment[BuilddBaseAlias.BIONIC] FAILED [ 98%]
  multipass/test_multipass_provider.py::test_launched_environment[BuilddBaseAlias.FOCAL] FAILED [ 98%]
  multipass/test_multipass_provider.py::test_launched_environment[BuilddBaseAlias.JAMMY] FAILED [ 98%]
  multipass/test_multipass_provider.py::test_launched_environment[BuilddBaseAlias.LUNAR] FAILED [ 99%]
  multipass/test_multipass_provider.py::test_launched_environment[BuilddBaseAlias.MANTIC] FAILED [ 99%]
  multipass/test_multipass_provider.py::test_launched_environment[BuilddBaseAlias.DEVEL] FAILED [100%]
[...]
  input = None, capture_output = True, timeout = None, check = True
  popenargs = (['multipass', 'info', 'itest-YXBIPNQK', '--format', 'json'],)
  kwargs = {'stderr': -1, 'stdout': -1, 'text': True}
  process = <Popen: returncode: 2 args: ['multipass', 'info', 'itest-YXBIPNQK', '--forma...>
  stdout = ''
  stderr = "info failed: ssh failed to authenticate: 'Socket error: disconnected'\n"
```

**Comment by mr-cal** [comment]
I think we're going to need to switch to the self-hosted runners, as now we're running out of space in [CI](https://github.com/canonical/craft-providers/actions/runs/6522004165/job/17711153090?pr=330#step:7:2575):
```
Cannot mkdir: No space left on device
```

**Comment by mr-cal** [comment]
Update, this is still blocked by a lack of disk space for testing.  We tried switching to self-hosted runners but ran into more problems.  We're going to look into moving our integration tests to spread in #430

**Comment by lengau** [comment]
You'll need to disable mantic on the macos tests until multipass on macos gets the ability to start mantic VMs. https://github.com/canonical/craft-providers/blob/main/tests/integration/multipass/test_multipass_provider.py#L55C49-L55C49

**Comment by syu-w** [comment]
Oh, it still only 23.04 atm

**Review by lengau — DISMISSED** [review]

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -33,6 +33,16 @@ jobs:
           cache: 'pip'
       - name: Configure environment
         run: |
+          echo "::group::Remove unnecessary packages"
```
Comment (mr-cal) [inline .github/workflows/tests.yaml:36]: You should be able to drop this from the PR now.  It is supposed to be for ` integration-tests-linux:` instead of `linters`.  Perhaps that's why it didn't initially work

**Review by mr-cal — APPROVED** [review]

**Review by lengau — APPROVED** [review]

**Comment by freyes** [comment]
now that this PR got merged, can we get a new release? (1.19.0 ? ), so it can get consumed by charmcraft and eventually this features makes its way into charmcraft 2.x ?

**Comment by syu-w** [comment]
@freyes the 1.19 is on the way to release.
